scilpy/io/utils.py

```diff
@@ -56,6 +55,64 @@ def check_tracts_same_format(parser, filename_list):
             parser.error('All tracts file must use the same format.')
 
 
+def assert_gradients_filenames_valid(parser, filename_list, gradient_format):
+    """
+    Validate if gradients filenames follow BIDS or MRtrix convention
+
+    Parameters
+    ----------
+    parser: parser
+        Parser.
+    filename_list: list
+        list of gradient paths.
+    gradient_format : str
+        Can be either fsl or mrtrix.
+
+    Returns
+    -------
+    """
+
+    valid_fsl_extensions = ['.bval', '.bvec']
+    valid_mrtrix_extension = '.b'
+
+    if isinstance(filename_list, str):
+        filename_list = [filename_list]
+
+    if gradient_format == 'fsl':
+        if len(filename_list) == 2:
+            filename_1 = filename_list[0]
+            filename_2 = filename_list[1]
+            basename_1, ext_1 = split_name_with_nii(filename_1)
+            basename_2, ext_2 = split_name_with_nii(filename_2)
+
+            if ext_1 == '' or ext_2 == '':
```

Comment:
Right there, we could handle both case.
Either files have an extension (`.bval` or .`bvec`), OR they can have an empty extension and have a basename being exactly (`bval` or `bvec`).

Only last question, how can we handle `.bval` .`bvec` as well as `.bvals` .`bvecs` (same thing with basename)
I think a lot people uses bvals and bval interchangeably 

What do you think @mdesco

Reply:
Personally I don't mind switching to an extension based convention. My only question is:
- In tractoflow and most simple organisation we name everything `dwi.nii.gz` `bvals` `bvecs` `rev_b0.nii.gz`
If we have an extension based convention, do we call them `bvals.bval` `bval.bval` or `scheme.bval` ? (or anything else)